### PR TITLE
ShipmentView: Receiving Shipment: Create Tests and View

### PR DESCRIPTION
**Pull request by vahidbazzaz**
In this PR, a receiving shipment table with sorting and filtering capabilities is added.

### CODE REVIEW

File: react/src/views/Transfers/ShipmentView/components/ShipmentReceivingContent.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  // Define columns
+  const columns = useMemo(
+    () => [
+      {
+        id: "labelIdentifier",
+        Header: "BOX #",
+        accessor: "labelIdentifier",
+        // eslint-disable-next-line react/no-unstable-nested-components
+        Cell: ({ row, column, state }: CellProps<any>) => {
```
Comment (HaGuesto) [inline react/src/views/Transfers/ShipmentView/components/ShipmentReceivingContent.tsx:32]: please remove this

File: react/src/views/Transfers/ShipmentView/components/ShipmentReceivingTable.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                  </Th>
+                ))}
+              </Tr>
+            ))}
+          </Thead>
+          <Tbody {...getTableBodyProps()}>
+            {rows.map((row) => {
+              prepareRow(row);
+              return (
+                <Tr {...row.getRowProps()}>
```
Comment (HaGuesto) [inline react/src/views/Transfers/ShipmentView/components/ShipmentReceivingTable.tsx:70]: please define how the row should look like

**Comment by vahidbazzaz** [comment]
[2.0 ShipmentView: Receiving Shipment: create tests + view](https://trello.com/c/DrBwz33S)

**Comment by pylipp** [comment]
## Functional review
I'm logged in as dev_coordinator@boxaid.org. I added a box to shipment 3 (source: Lesvos (my base), target: Samos) and Finalized&sent it. Via playground I did "start receiving shipment". When I refresh `transfers/shipments/3` I see the receiving UI but acc. to [this doc](https://docs.google.com/spreadsheets/d/1-L_h3u6fakOoy6P62fP0dWrQ6IbSfjKvtBjmtmDqZ74/edit#gid=0) I should see "View Shipment" UI since I'm sender.

**Comment by vahidbazzaz** [comment]
@HaGuesto @pylipp currently the total number of boxes does not correctly display, but once this [trello card](https://trello.com/c/AhtWx3LO/1183-20-enable-clearly-showing-shipment-details-of-completed-shipments) is done, this issue should be resolved in the upcoming PR.

**Comment by pylipp** [comment]
@vahidbazzaz I went for another functional test. Below is the proposed figma screen as of today (top), and a screenshot from the current implementations:

2023-04-20-115349_176x366_scrot

2023-04-20-115418_353x608_scrot

I have the following remarks:
1. the proposed title is "Receiving shipment", not "Receive shipment"
2. the column "Product" in the table header is not aligned with the actual product cells in the table below
3. the item count has a smaller size than the product name, and there is a whitespace between
4. in the "Remaining" box there should be no box icon acc. to the figma screen

I assume the Figma screen reflects the latest state of the UI as intended by @aerinsol , hence @vahidbazzaz please update the UI implementation accordingly in the next PR for the 2nd iteration **(not in this PR).**

> I'm logged in as [dev_coordinator@boxaid.org](mailto:dev_coordinator@boxaid.org). I added a box to shipment 3 (source: Lesvos (my base), target: Samos) and Finalized&sent it. Via playground I did "start receiving shipment". When I refresh transfers/shipments/3 I see the receiving UI but acc. to [this doc](https://docs.google.com/spreadsheets/d/1-L_h3u6fakOoy6P62fP0dWrQ6IbSfjKvtBjmtmDqZ74/edit#gid=0) I should see "View Shipment" UI since I'm sender.

I can confirm that the above issue has been resolved.

I'm leaving a code review to @HaGuesto :sunglasses:

**Review by pylipp — APPROVED** [review]

**Comment by aerinsol** [comment]
@vahidbazzaz I don't mind the box icon for now in receiving, but @pylipp other points are completely on point. Also, why is the QR scanner icon still there? I thought it had been merged into the menu?

**Pull request merged**